aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -74,26 +75,51 @@ void index_select_add<float>(const Tensor &select_indices,
                              const Tensor &add_indices,
                              const Tensor &src,
                              Tensor &output,
-                             const Tensor& offsets) {
+                             const Tensor& offsets,
+                             bool include_last_offset) {
   int64_t ddim = src.size(1);
   auto src_data = src.data_ptr<float>();
   auto select_indices_data = select_indices.data_ptr<int64_t>();
   auto output_data = output.data_ptr<float>();
 
   if (isFastPathIndexSelect(src, output)) {
-    caffe2::EmbeddingLookupIdx(
-      /*block_size=*/ddim,
-      /*output_size=*/offsets.numel(),
-      /*index_size=*/select_indices.numel(),
-      /*data_size=*/src.size(0),
-      /*input=*/src_data,
-      /*indices=*/select_indices_data,
-      /*offsets=*/offsets.data_ptr<int64_t>(),
-      /*weights=*/nullptr,
-      /*scale_bias=*/nullptr,
-      /*normalize_by_lengths=*/false,
-      /*out=*/output_data
-    );
+    if (include_last_offset) {
+      auto output_size = offsets.numel() - 1;
+      at::parallel_for(
+          0, output_size, 1, [&](int64_t start_idx, int64_t end_idx) {
+            caffe2::EmbeddingLookupIdx(
+                /*block_size=*/ddim,
+                /*output_size=*/end_idx - start_idx,
+                /*index_size=*/select_indices.numel(),
+                /*data_size=*/src.size(0),
+                /*input=*/src_data,
+                /*indices=*/select_indices_data,
```

Comment:
We can pass select_indices_data + offsets.data_ptr<int64_t>()[start_idx] and, for index_size, offsets.data_ptr<int64_t>()[end_idx] - offsets.data_ptr<int64_t>()[start_idx] . This should enable error handling (see comment below).